Upgrade Django 3.2 -> 4.2

martynasracct opened the pull request:
**Summary:**
Upgrade Django 3.2 -> 4.2

Main changes:
- Model `save()` methods now includes fields in `update_fields`, if `update_fields` is used when saving.
- Move `DeleteView` delete logic to `form_valid` methods.
- Call `DeleteView` using POST, if view doesn't have HTML template. If it has - GET request will show confirmation page.

**Ticket:**
https://github.com/atviriduomenys/katalogas/issues/1812

martynasracct committed:
Upgrade Django 3.2 -> 4.2

Fix model `.save()` methods to work with .save(update_fields=[...])
Fix DeleteView custom delete logic

varzgalys commented:
"Call DeleteView using POST, if view doesn't have HTML template. If it has - GET request will show confirmation page." - isn't that breaking UI part?

varzgalys commented on tests/datasets/test_views.py:
```diff
@@ -4251,3 +4257,64 @@ def test_dataset_rdf_download__datas_service(app: DjangoTestApp):
+        dataset = DatasetFactory()
+
+        app.post(reverse("dataset-delete", args=[dataset.pk]))
+        assert not Dataset.objects.exists()
```
Please assert using PK, as this could lead to test Fail with https://github.com/atviriduomenys/katalogas/pull/1841 🙂 


```suggestion
        app.post(reverse("dataset-delete", args=[dataset.pk]))
        assert not Dataset.objects.filter(pk=dataset.pk).exists()
```

martynasracct replied:
Fully agree. Fixed in 797ada5082a2f6ca2ed2730bd11268a8a28e03cc

varzgalys commented on tests/datasets/test_views.py:
```diff
@@ -4251,3 +4257,64 @@ def test_dataset_rdf_download__datas_service(app: DjangoTestApp):
+
+        app.post(reverse("dataset-request-remove", args=[dataset.pk, request_object.pk]))
+
+        assert Dataset.objects.exists()
```
assert by id here as well
```suggestion
        assert Dataset.objects.filter(pk=dataset.pk).exists()
```

martynasracct replied:
Fixed in 797ada5082a2f6ca2ed2730bd11268a8a28e03cc

varzgalys commented on vitrina/tasks/models.py:
```diff
@@ -105,6 +105,8 @@ def __str__(self):
     def save(self, *args, **kwargs):
         self.due_date = get_due_date()
+        if update_fields := kwargs.get("update_fields"):
+            kwargs["update_fields"] = {"due_date"}.union(update_fields)
```
Optional
```suggestion
            kwargs["update_fields"] = update_fields | {"due_date"}
```

martynasracct replied:
Fixed in 797ada5082a2f6ca2ed2730bd11268a8a28e03cc

varzgalys commented on vitrina/uapi/models.py:
```diff
@@ -91,10 +91,14 @@ class Meta:
     def save(self, *args, **kwargs) -> None:
         self.codename = self.get_codename(self.title)
 
+        if (
+            update_fields := kwargs.get("update_fields")
+        ) is not None and "title" in update_fields:
```
```suggestion
        if update_fields := kwargs.get("update_fields") and "title" in update_fields:
```

Fits into 120 char and is None safe

Struchid replied:
fyi, ^ will need to put this part into brackets to work:
`update_fields := kwargs.get("update_fields")`

martynasracct replied:
Agree with both of you. Fixed in 797ada5082a2f6ca2ed2730bd11268a8a28e03cc

Struchid commented on vitrina/datasets/templates/vitrina/datasets/datatable.html:
```diff
@@ -85,10 +85,15 @@
+                    {% csrf_token %}
+                    <button type="submit"
+                            class="button is-link is-small is-pulled-right"
+                            onclick="return confirm('Ar jūs tikrai norite trinti šį šaltinį?')"
```
I might be wrong, but I think the word used does not have the definition of "deleting something":

```suggestion
                            onclick="return confirm('Ar jūs tikrai norite ištrinti šį šaltinį?')"
```

What my assumptions are based on:
- https://www.zodynas.lt/terminu-zodynas/t/trinti

martynasracct replied:
Tbf, I left this text as it was. But yeah, "ištrinti" is better. Hmm, it would be even better if I also translated this text 🤔

martynasracct replied:
Fixed in f5897e7ae6dcade5c131943ebbda3dbead15ebcd

Struchid commented on vitrina/datasets/templates/vitrina/datasets/detail.html:
Same for occurrences in this file if you agree with the previous comment

martynasracct replied:
Fixed in f5897e7ae6dcade5c131943ebbda3dbead15ebcd

Struchid commented on vitrina/orgs/views.py:
```diff
@@ -1279,17 +1281,15 @@ def get_success_url(self):
-        pre_representative_delete(obj)
-        return super().delete(request, *args, **kwargs)
+    def form_valid(self, form: BaseForm) -> HttpResponse:
+        if self.object.organization:
+            Dataset.objects.filter(
+                organization_id=self.kwargs.get("organization_id"),
```
**Optional:** Would save a query if the `organization_id` is not in `kwargs` (assuming it could be nullable, because of the `.get` used)
```suggestion
        if self.object.organization and (organization_id := self.kwargs.get("organization_id")):
            Dataset.objects.filter(
                organization_id=organization_id,
```

martynasracct replied:
Fixed in 797ada5082a2f6ca2ed2730bd11268a8a28e03cc

Struchid reviewed (approved):
Few minor comments, otherwise LGTM 🚀

martynasracct committed:
Fixes after PR review

martynasracct committed:
Fix and translate delete confirmations, fix translations

martynasracct commented:
> "Call DeleteView using POST, if view doesn't have HTML template. If it has - GET request will show confirmation page." - isn't that breaking UI part?

Previously all `DeleteView` called GET method, but some `DeleteView.get` methods called `post` and others had HTML template.

Now, if `DeleteView` has template - it does GET request, if not - it directly calls POST.

martynasracct committed:
Fix broken test

martynasracct committed:
Add CSRF_TRUSTED_ORIGINS to settings

martynasracct committed:
Fixes for not working generate-pdf view

Fix crash when page loads, because of missing arguments for url reverse
Add breadcrumbs and page title
Fix issue when after generating pdf both success and error messages are showed

martynasracct committed:
Fixes after rebase

martynasracct committed:
Move tests in same file

martynasracct committed:
Fix failing test after rebase
